### PR TITLE
Update to RTOS page and Kernel namespace documentation

### PR DESCRIPTION
**Pull request by SenRamakri**
This PR contains following docs changes:
1. Added Kernel page
2. Updated RTOS page

### CODE REVIEW

**Comment by SenRamakri** [comment]
@AnotherButler - Can you please add you, @sg- , @kegilbert as reviewers for this PR?

**Comment by AnotherButler** [comment]
Please complete the JIRA task by doing the following:

- Remove all references to CMSIS-RTOS both on this page and throughout this folder.
- Remove all reference to RTX both on this page and throughout this folder.
- Ensure the error codes on this page are included in the Doxygen, and then delete them here.
- Include RTOS design patterns and use cases.
- Include two Mbed OS shared queues.

**Comment by SenRamakri** [comment]
@AnotherButler - Mbed OS shared queue is captured in https://os.mbed.com/docs/v5.9/reference/eventqueue.html
I don't think the shared queue example belongs in top-level RTOS page.

Also, can you please clarify on what it means by "Include RTOS design patterns and use cases."

**Comment by AnotherButler** [comment]
What are some general best practices or typical code patterns developers use when using these APIs? At a high level, when do developers use RTOS APIs? When don't they?

**Comment by SenRamakri** [comment]
@AnotherButler - I think we can possibly include that, I got confused by design pattern as it can mean using these on some software design perspective.

**Comment by SenRamakri** [comment]
@AnotherButler - I looked into this more. As of now, the RTOS page does have section "RTOS APIs" which lists all the objects under RTOS layer. And each of those takes the user to specific page explaining that object. And for each of those objects we have APIs examples/code snippets as required. The one which was missing before was Kernel APIs and I have added them as well. Please review if its not what you are looking for.

File: docs/reference/api/rtos/Kernel.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Kernel interface functions
+
+The Kernel namespace implements functions to read RTOS information. It implements one function to read the current RTOS kernel millisecond tick count.
+
+### Kernel namespace reference
+
```
Comment (AnotherButler) [inline docs/reference/api/rtos/Kernel.md]: Note: This link doesn't work for me. Is there a code dependency?

File: docs/reference/api/rtos/rtos.md [hunk 2]
```diff
@@ -1,14 +1,16 @@
 <h2 id="rtos-api">RTOS overview</h2>
 
-The Arm Mbed RTOS is a C++ wrapper over the Keil RTX code. For more information about Keil RTX, check [the Keil CMSIS-RTOS tutorial](https://github.com/ARM-software/CMSIS/raw/master/CMSIS/Documentation/RTX/CMSIS_RTOS_Tutorial.pdf) and [the element14 introduction to Keil RTX](https://www.element14.com/community/docs/DOC-46650/l/arm-keil-rtx-real-time-operating-system-overview). You can use these resources as a general introduction to RTOS principles; it is important to be familiar with the concepts behind an RTOS in order to understand this guide.
+Arm Mbed OS provides C++ APIs to manage RTOS objects such as thread, synchronization objects and timer. It also provides interfaces for attaching an application-specific idle hook function and to read the tick count from OS. The Arm Mbed RTOS layer also handles RTOS errors and reports them into the Mbed OS error handling system.
```
Comment (sg-) [inline docs/reference/api/rtos/rtos.md]: > Arm Mbed OS provides C++ APIs to manage

is the same as saying The Arm Mbed RTOS is a C++ wrapper over the Keil RTX code just in a different way. How about something like along the lines of the mbed os rtos capabilities include things such as ...

File: docs/reference/api/rtos/rtos.md [hunk 3]
```diff
@@ -23,6 +25,8 @@ The RTOS APIs handle creation and destruction of threads in Arm Mbed OS 5, as we
 - [RtosTimer](/docs/development/reference/rtostimer.html): A deprecated class used to control timer functions in the system.
 - [EventFlags](/docs/development/reference/eventflags.html): An event channel that provides a generic way of notifying other threads about conditions or events.
 - [Event](/docs/development/reference/event.html): The queue to store events, extract them and excute them later.
+- [ConditionVariable](/docs/development/reference/conditionvariable.html): The ConditionVariable class provides a mechanism to safely wait for or signal state changes.
```
Comment (sg-) [inline docs/reference/api/rtos/rtos.md]: this sounds similar to how event flag is defined. How is it different?

File: docs/reference/api/rtos/rtos.md [hunk 4]
```diff
@@ -51,21 +55,6 @@ Each `Thread` can wait for signals and be notified of events:
...
-- `osErrorTimeoutResource`:  a specified resource was not available within the timeout period.
-- `osErrorISR`: the function cannot be called from interrupt service routines (ISR).
-- `osErrorISRRecursive`: function called multiple times from ISR with same object.
-- `osErrorPriority`: system cannot determine priority or thread has illegal priority.
-- `osErrorNoMemory`: system is out of memory; it was impossible to allocate or reserve memory for the operation.
-- `osErrorValue`: value of a parameter is out of range.
-- `osErrorOS`: unspecified RTOS error - runtime error but no other error message fits.
+##### Status and Error codes
+
+The Arm Mbed RTOS layer handles RTOS errors and report them using the Mbed OS error handling system. Please see the list of error codes in the [error handling](/docs/development/reference/error-handling.html) documentation for more information on RTOS errors reported.
```
Comment (sg-) [inline docs/reference/api/rtos/rtos.md]: dont describe layers, The RTOS errors codes are documented as part of OS errors...

**Comment by SenRamakri** [comment]
@sg-, @kegilbert - Please review as I have updated with review comments.

File: docs/reference/api/rtos/rtos.md [hunk 5]
```diff
@@ -1,14 +1,18 @@
 <h2 id="rtos-api">RTOS overview</h2>
 
-The Arm Mbed RTOS is a C++ wrapper over the Keil RTX code. For more information about Keil RTX, check [the Keil CMSIS-RTOS tutorial](https://github.com/ARM-software/CMSIS/raw/master/CMSIS/Documentation/RTX/CMSIS_RTOS_Tutorial.pdf) and [the element14 introduction to Keil RTX](https://www.element14.com/community/docs/DOC-46650/l/arm-keil-rtx-real-time-operating-system-overview). You can use these resources as a general introduction to RTOS principles; it is important to be familiar with the concepts behind an RTOS in order to understand this guide.
+Mbed-OS RTOS provide C++ APIs to manage objects like thread, synchronization objects and timer. It also provides interfaces for attaching application
```
Comment (kegilbert) [inline docs/reference/api/rtos/rtos.md]: Minor nit:
Change to "The Mbed OS RTOS provides C++ APIs to manage objects like threads, synchronization objects, and timers."

Comment (SenRamakri) [reply docs/reference/api/rtos/rtos.md]: @kegilbert - Change done, please review.

**Pull request merged**